src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(stateClone, action.extraData);
+      result.push({ ...stateClone });
```

Comment:
Instead of performing the 'push' operation inside every condition, you can do it just once after the action handling code. This will adhere to the 'Don't Repeat Yourself' principle.